pgjdbc/src/main/java/org/postgresql/jdbc/ResultWrapper.java

```diff
@@ -47,17 +51,24 @@ public long getInsertOID() {
     return next;
   }
 
-  public void append(ResultWrapper newResult) {
+  public ResultWrapper append(ResultWrapper newResult) {
+    if (commandType == SqlCommandType.SET) {
+      return newResult;
+    }
+    if (newResult.commandType == SqlCommandType.SET) {
+      return this;
+    }
```

Comment:
Could you add a comment clarifying the reason behind these conditions?

Reply:
I hope method documentation is ok